modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java

```diff
@@ -252,7 +252,7 @@
     private final BinaryContext ctx;
 
     /** Binary metadata handler. */
-    private final JdbcBinaryMetadataHandler metaHnd;
+    private JdbcBinaryMetadataHandler metaHnd;
```

Comment:
Are you sure metaHnd is not invoked in concurrent ?

Reply:
JDBC Connection is not thread safe by specification.